src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.UnitConverter/Main.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Windows;
+using System.Windows.Input;
+using ManagedCommon;
+using UnitsNet;
+using Wox.Plugin;
+
+namespace Community.PowerToys.Run.Plugin.UnitConverter
+{
+    public class Main : IPlugin, IPluginI18n, IContextMenu, IDisposable
+    {
+        public string Name => Properties.Resources.plugin_name;
+
+        public string Description => Properties.Resources.plugin_description;
+
+        private PluginInitContext _context;
+        private static string _icon_path;
+        private bool _disposed;
+        private static readonly QuantityType[] _included = new QuantityType[]
+        {
+            QuantityType.Acceleration,
+            QuantityType.Angle,
+            QuantityType.Area,
+            QuantityType.Duration,
+            QuantityType.Energy,
+            QuantityType.Information,
+            QuantityType.Length,
+            QuantityType.Mass,
+            QuantityType.Power,
+            QuantityType.Pressure,
+            QuantityType.Speed,
+            QuantityType.Temperature,
+            QuantityType.Volume,
+        };
+
+        private readonly CultureInfo _currentCulture = CultureInfo.CurrentCulture;
+        private readonly int _roundingFractionalDigits = 4;
+
+        public void Init(PluginInitContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(paramName: nameof(context));
+            }
+
+            _context = context;
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+        }
+
+        public List<Result> Query(Query query)
+        {
+            if (query == null)
+            {
+                throw new ArgumentNullException(paramName: nameof(query));
+            }
+
+            string[] split = query.Search.Split(' ');
+
+            InputInterpreter.ShorthandFeetInchHandler(ref split, _currentCulture);
+            InputInterpreter.InputSpaceInserter(ref split);
+
+            if (split.Length < 4 || split.Length > 4)
+            {
+                // deny any other queries than:
+                // 10 ft in cm
+                // 10 ft to cm
+                return new List<Result>();
+            }
+
+            InputInterpreter.DegreePrefixer(ref split);
```

Comment:
We should move out InputInterpreter logic out of `Query` into InputInterpreter.Parse function. Ideally, we should have an intermediate structure between Parse and Execute. That is `Parse` returns the structure and `UnitHandler.Convert` receives that structure and returns result.